src/Chapter3.hs

```diff
@@ -1006,7 +1122,78 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
-
+--Assuming data for each player (monster/knight)
+newtype PlayerHealth = PlayerHealth Int
+newtype PlayerAttack = PlayerAttack Int
+newtype PlayerDefense = PlayerDefense Int
+
+--All possible actions
+data BattleAction = Attack | DrinkPotion Int | CastSpell Int | MonsterAttack | MonsterRun
```

Comment:
Ideally, you want to have two separate data type `FighterAction` and `MonsterAction` so it won't be possible to mix `MonsterRun` and `DrinkPotion` in a single list